reya_data_feed/resources/market.py

```diff
@@ -0,0 +1,164 @@
+        self._market_orders = MarketOrdersResource(socket)
+    
+    @property
+    def all_markets(self) -> 'AllMarketsResource':
+        """Access the all markets data resource."""
+        return self._all_markets
+    
+    def market_data(self, market_id: str) -> 'MarketDataSubscription':
+        """Get market data for a specific market.
+        
+        Args:
+            market_id: The ID of the market.
+            
+        Returns:
+            A subscription object for the specified market data.
+        """
+        return self._market_data.for_market(market_id)
+    
+    def market_orders(self, market_id: str) -> 'MarketOrdersSubscription':
+        """Get orders for a specific market.
+        
+        Args:
+            market_id: The ID of the market.
+            
+        Returns:
+            A subscription object for the specified market orders.
+        """
+        return self._market_orders.for_market(market_id)
+
+class AllMarketsResource(SubscribableResource):
+    """Resource for accessing all markets data."""
+    
+    def __init__(self, socket: 'ReyaSocket'):
+        """Initialize the all markets data resource.
+        
+        Args:
+            socket: The WebSocket connection to use for this resource.
+        """
+        super().__init__(socket)
+        self.path = "/api/trading/markets/data"
+    
+    def subscribe(self, batched: bool = False) -> None:
+        """Subscribe to all markets data.
+        
+        Args:
+            batched: Whether to receive updates in batches.
+        """
+        self.socket.send_subscribe(channel=self.path, batched=batched)
+    
+    def unsubscribe(self) -> None:
+        """Unsubscribe from all markets data."""
+        self.socket.send_unsubscribe(channel=self.path)
+
+class MarketDataResource(SubscribableParameterizedResource):
+    """Resource for accessing market data."""
+    
+    def __init__(self, socket: 'ReyaSocket'):
+        """Initialize the market data resource.
+        
+        Args:
+            socket: The WebSocket connection to use for this resource.
+        """
+        super().__init__(socket, "/api/trading/market/{market_id}/data")
+    
+    def for_market(self, market_id: str) -> 'MarketDataSubscription':
+        """Create a subscription for a specific market's data.
+        
+        Args:
+            market_id: The ID of the market.
+            
+        Returns:
+            A subscription object for the specified market data.
+        """
+        return MarketDataSubscription(self.socket, market_id)
+
+class MarketOrdersResource(SubscribableParameterizedResource):
+    """Resource for accessing market orders."""
+    
+    def __init__(self, socket: 'ReyaSocket'):
+        """Initialize the market orders resource.
+        
+        Args:
+            socket: The WebSocket connection to use for this resource.
+        """
+        super().__init__(socket, "/api/trading/market/{market_id}/orders")
```

Comment:
i think we're not currently using this endpoint in production (can also double check on DD for more confidence)

should be able to just rename the existing endpoint to /trades to have consistent terminology

Reply:
Fixed !